parser-utils/src/lib.rs

```diff
@@ -8,6 +8,14 @@
 // CONDITIONS OF ANY KIND, either express or implied. See the License for the
 // specific language governing permissions and limitations under the License.
 
+extern crate combine;
+
+use self::combine::ParseResult;
+use self::combine::combinator::{Expected, FnParser};
+
+// Nothing about this is specific to the type of parser.
+pub type ResultParser<O, I> = Expected<FnParser<I, fn(I) -> ParseResult<O, I>>>;
```

Comment:
What is the intended use for this? Worth adding more documentation here? This is also copied in `parser_util.rs`.